src/libsystemd-network/dhcp-option.c

```diff
@@ -284,6 +287,102 @@ int dhcp_option_append(DHCPMessage *message, size_t size, size_t *offset,
         return -ENOBUFS;
 }
 
+static int dhcp_option_append_long(DHCPMessage *message, size_t size, size_t *offset,
+                                   uint8_t overload,
+                                   uint8_t code, size_t optlen, const uint8_t *optval) {
+        const bool use_file = overload & DHCP_OVERLOAD_FILE;
+        const bool use_sname = overload & DHCP_OVERLOAD_SNAME;
+        size_t available, to_write;
+        int r;
+
+        assert(message);
+        assert(offset);
+
+        /* If *offset is in range [0, size), we are writing to ->options,
+         * if *offset is in range [size, size + sizeof(message->file)) and use_file, we are writing to ->file,
+         * if *offset is in range [size + use_file*sizeof(message->file),
+         *                         size + use_file*sizeof(message->file) + sizeof(message->sname))
+         * and use_sname, we are writing to ->sname.
+         */
+
+        while (optlen) {
```

Comment:
we generally do not rely on C's downgrade-to-bool feature when checking if numeric values or zero/non-zero. We use it only if the variable is used like a boolean in most ways.

or in other words, write this as `while (optlen > 0) {`

Reply:
Agreed, changed.